modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/actions/PreIssueAccessTokenTestCase.java

```diff
@@ -0,0 +1,470 @@
+
+        super.init(TestUserMode.TENANT_USER);
+
+        oAuth2Service = this.new OAuth2ServiceHelper();
+        scim2RestClient = new SCIM2RestClient(serverURL, tenantInfo);
+
+        setSystemproperties();
+
+        /* Pre Requisites */
+
+        ApplicationResponseModel application = oAuth2Service.addApplication(PASSWORD_GRANT_TYPE);
+
+        // creates systems APIs
+        if (!CarbonUtils.isLegacyAuthzRuntimeEnabled()) {
+            oAuth2Service.authorizeSystemAPIs(application.getId(), new ArrayList<>(Arrays.asList(SCIM2_USERS_API, ACTIONS_API,
+                    APPLICATION_MANAGEMENT_API, API_RESOURCE_MANAGEMENT_API)));
+        }
+
+        // creates domain APIs
+        List<String> customScopes = new ArrayList<>();
+        Collections.addAll(customScopes, CUSTOM_SCOPE_1, CUSTOM_SCOPE_2, CUSTOM_SCOPE_3);
+        String businessAPIId = oAuth2Service.createDomainAPIs(EXTERNAL_SERVICE_NAME, EXTERNAL_SERVICE_URI, customScopes);
+
+        // authorizes domain APIs
+        oAuth2Service.authorizeDomainAPIs(application.getId(), businessAPIId, customScopes);
+
+        // creates users and roles
+        List<Permission> permissions = addUserWithRole(application.getId(), customScopes);
+
+        // creates pre issue access token action type
+        createPreIssueAccessTokenAction();
+
+        // requests access token
+        OpenIDConnectConfiguration oidcConfig = oAuth2Service.getOIDCInboundDetailsOfApplication(application.getId());
+        String tenantedTokenURI = getTenantQualifiedURL(OAuth2Constant.ACCESS_TOKEN_ENDPOINT, tenantInfo.getDomain());
+        oAuth2Service.requestAccessToken(oidcConfig.getClientId(), oidcConfig.getClientSecret(), tenantedTokenURI, TEST_USER, ADMIN_WSO2, permissions);
+    }
+
+    // TODO: check what is wrong here
+//    @AfterClass(alwaysRun = true)
+//    public void atEnd() throws Exception {
+//        restClient.deleteV2Role(roleID);
+//        deleteApp(appId);
+//        scim2RestClient.deleteUser(userID);
+//        consumerKey = null;
+//        consumerSecret = null;
+//        appId = null;
+//        restClient.closeHttpClient();
+//    }
+
+    @DataProvider(name = "getNewCustomScope")
+    public Object[][] getNewCustomScope() {
+        String[] scopesArray = new String[]{"new_test_custom_scope_1", "new_test_custom_scope_2", "new_test_custom_scope_3"};
+
+        return new Object[][]{
+                {"scope", scopesArray},
+        };
+    }
+
+    @Test(groups = "wso2.is", description = "Check if the added scope is present in the access token",
+            dataProvider = "getNewCustomScope")
+    public void testTokenScopeAddOperation(String scope, String[] newScopeArray) throws Exception {
+        JWTClaimsSet jwtClaims = extractJwtClaims(accessToken);
+
+        String scopeString = jwtClaims.getStringClaim(scope);
+        String[] scopes = scopeString.split("\\s+");
+
+        Assert.assertTrue(ArrayUtils.contains(scopes, newScopeArray[0]));
```

Comment:
Since we already know that the scope is `new_test_custom_scope_1 `, can't we directly check that string here without getting it through data provider.
If so we can get rid off from the data provider in this test method

Reply:
Fixed with the commit a1f9cd1ba7fe4e68789e6f894da50c4b3bbf3d23